OneCommunityGlobal/HighestGoodNetworkApp
# Olga add tool page

# Description
Components for Add tool form per the wireframe below:

![Add tool wireframe](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/88c9f7e1-de24-49de-ab04-dc6c881a9c51)


## Main changes explained:
- Created components: AddTool.jsx, AddToolForm.jsx, PhoneInput.jsx, DragAndDrop.jsx and corresponding css file

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. check into dev branch in the backend and start the backend
4. in code go to AddToolForm.jsx and comment in lines 64-70 (updatedFormData and console statement)
5. go to http://localhost:3000/BMdashboard/tools/add (log in if you are prompted to do so)
6. check the design in different screen sizes and functionality of the form 
7. check that when clicking submit all the form data is logged correctly in console
8. check that when clicking cancel the form is cleared
9. check that you can upload a picture by dragging and by clicking on the upload area
10. check that the total price is displayed correctly


## Note:
1. Note that the functionality of the Add forms has changed/simplified to adding a new tool "type" to the buildingInventoryTypes db collection, therefore the form is slightly different from the wireframe and does not include 2nd input field (2 in the picture above).
2. There is currently no validation on the form, this will be a different PR.
